File: server/lib/ria/guitars/guitar.ex
defmodule Ria.Guitars.Guitar do
  use Ecto.Schema
  import Ecto.Changeset

  schema "guitars" do
    field :brand, :string
    field :img_url, :string
    field :prod_year, :integer
    field :type, :string

    timestamps()
  end

  @doc false
  def changeset(guitar, attrs) do
    guitar
    |> cast(attrs, [:type, :brand, :img_url, :prod_year])
    |> validate_required([:type, :brand, :prod_year])
  end
end

File: client/src/app/resources/guitar.ts
export class Guitar {
  id?: number;
  imgUrl?: string;
  type: string;
  brand: string;
  prodYear: number;
  constructor(
    type: string,
    brand: string,
    prodYear: number,
    imgUrl?: string,
    id?: number
  ) {
    this.type = type;
    this.brand = brand;
    this.imgUrl = imgUrl;
    this.prodYear = prodYear;
    this.id = id;
  }
}
